common-service-core/src/main/java/org/zowe/apiml/util/MapUtils.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * This program and the accompanying materials are made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Copyright Contributors to the Zowe Project.
+ */
+
+package org.zowe.apiml.util;
+
+import java.util.*;
+import java.util.function.Consumer;
+
+public class MapUtils {
+
+    public Map<String, String> flattenMap(String rootKey, Map<String, Object> collection) {
+        if (collection == null || collection.isEmpty()) {
+            return Collections.emptyMap();
+        }
+
+        Map<String, String> result = new HashMap<>();
+
+        Map<Class, Consumer<Map.Entry<String, Object>>> actionMap = new LinkedHashMap<>();
```

Comment:
What about make this mapping in static part. For rootKey of course use BiFunction. In this case the map is constructed all times.

Reply:
check out the latest impl, I've modified the code to do this